class-udm-updater.php

```diff
@@ -152,10 +168,12 @@ private function get_puc_dir() {
 	public function auto_update_plugin($update, $item) {
 
 		if (!isset($item->slug) || $item->slug != $this->slug || !$this->allow_auto_updates) return $update;
-		
-		$options = $this->get_option($this->option_name);
 
-		if (is_array($options) && !empty($options['auto_update'])) $update = true;
+		$option_auto_update_settings = $this->get_option('auto_update_plugins');
```

Comment:
@andtown `$this->get_option()` will call `get_site_option()` or `get_option()` depending on whether the install is single-site or multisite. Is that actually what's wanted? i.e. the WP option differs in where it is stored??

Reply:
@DavidAnderson684 `auto_update_plugins` will always be stored either in the `wp_sitemeta` table if multisite is in use or `wp_option` table if single-site. Since `get_site_option()` has its own multisite check and it basically does the same as `$this->get_option()`, we could just use `get_site_option()` everywhere in the code to get meta/option values from multi/single-site installation or we could also modify the `$this->get_option()` into wrapper for `get_site_option()` and remove the multisite check in it. Looking at how the `get_site_option()` works on the inside, it first calls a wrapper named `get_network_option()` in which the multisite check gets conducted, it then gets a meta/option value from the `wp_sitemeta` or `wp_options` table depending on whether single or multisite is in use. So yes, in this case, I'm fairly sure that calling `$this->get_option` works too, but let me know if you have other thoughts or if I've possibly misunderstood the intent of the issue here?

Reply:
@andtown Since WP core uses `get_site_option()` everywhere for that option, we should do the same. Just because that falls through to the same result as `get_option()` if you're not on multisite doesn't make it a good idea for us to do something different.